src/index.ts

```diff
@@ -76,7 +94,7 @@ export interface UploadConfig {
    * This will cause the upload to fail if the current generation of the remote
    * object does not match the one provided here.
    */
-  generation?: number;
+  generation?: QueryParameters['ifGenerationMatch'];
```

Comment:
I don't quite understand what this type is doing :) Couldn't this just be a `number` still?

Reply:
I thought it was smarter to re-use from a single source of definitions, but I don't know! What do you think?

Reply:
I'm kind of with @JustinBeckwith on this one, I think if `QueryParameters['ifGenerationMatch']` is a `number`, it reads a bit better to just keep the basic types (until we have a compelling reason, e.g., QueryParameters['foo'] becomes an enumeration.

Reply:
That's cool, I reverted back 👍